client/app/expenses/hooks/useUpdateExpense.ts

```diff
@@ -26,7 +26,7 @@ export const useUpdateExpense = (defaultData: ExpenseType) => {
   const handleSuccess = () => {
     toast({ variant: 'success', title: 'Successfully updated the expense!' })
     queryClient.invalidateQueries({
-      queryKey: ['expenses', { id: defaultData.id }],
+      queryKey: ['expenses'],
```

Comment:
I think you still need the ID here because you don't want to update the other GET/expenses/ID. Here, you want to delete the cache of the expense itself and the expense list (not other expense id's data)

Reply:
Fixed!
https://github.com/HAK2024/fairy-share/pull/113/commits/1472fee8c826a4f663622bd443c7a5218a0d83d6